doc/source/algorithms.rst

```diff
@@ -163,6 +163,7 @@ The following is an example that shows how to use this function:
 The result has several fields, among which the following are most useful:
 
 * ``parents[i]``:  the parent vertex of the i-th vertex. The parent of each source vertex is itself.
+* ``hasparent[i]``:  ``true`` if the i-th vertex has a parent, and ``false`` otherwise. When ``hasparent[i] == false``, it means that the vertex at index ``i`` isn't reachable from any source. Note that ``hasparent[i] == true`` for all source vertices.
```

Comment:
Yeah, I agree with you there. I'm just wondering whether this should be a local decision for `hasparents` (which I'll be happy to change), or whether it warrants opening an issue for the `parents` attribute as well